plugins/permission-node/src/ServerPermissionClient.ts

```diff
@@ -130,6 +142,44 @@ export class ServerPermissionClient implements PermissionsService {
     return options;
   }
 
+  #decideBasedOnPrincipalScope(
+    requests: Array<QueryPermissionRequest | AuthorizePermissionRequest>,
+    options?: PermissionsServiceRequestOptions,
+  ): DefinitivePolicyDecision[] | undefined {
+    if (!options || !('credentials' in options)) {
+      return undefined;
+    }
+
+    const credentials = options.credentials;
+    if (
+      !this.#auth.isPrincipal(credentials, 'service') ||
+      !credentials.principal.scope
+    ) {
+      return undefined;
+    }
+
+    const { permissionNames, permissionAttributes } =
+      credentials.principal.scope;
+
+    return requests.map(query => {
+      if (permissionNames && !permissionNames.includes(query.permission.name)) {
+        return { result: AuthorizeResult.DENY };
+      }
+      if (permissionAttributes) {
+        for (const [key, value] of Object.entries(permissionAttributes)) {
+          if (
+            (
+              query.permission.attributes as Record<string, unknown> | undefined
+            )?.[key] !== value
+          ) {
+            return { result: AuthorizeResult.DENY };
+          }
+        }
```

Comment:
then this one will become:

```suggestion
       if(permissionAttributes?.action && permissionAttributes.action !== query.permission.attributes.action) {
         return { result: AuthorizeResult.DENY };
       }
```

Reply:
Updated! Not exactly like that, because we need to support more than one possible value at once, but